ZenPacks/zenoss/Microsoft/Windows/WinService.py

```diff
@@ -97,17 +96,9 @@ def monitored(self):
             return self.monitor
 
         # 2 - Check what our template says to do.
-        template = self.getRRDTemplate()
-        if template:
-            datasource = template.datasources._getOb('DefaultService', None)
-            if datasource:
-                if self.getMonitored(datasource):
-                    return True
-            # 3 - Allow for other datasources to be specified.
-            for datasource in template.getRRDDataSources():
-                if datasource.id != 'DefaultService' and hasattr(datasource, 'startmode'):
-                    if self.getMonitored(datasource):
-                        return True
+        datasource = self.getMonitoredDataSource()
```

Comment:
This should be an instance variable so we can reference it later in getFailSeverity().  That's going to be a lot of looping on each cycle for each and every service that's monitored.  A single device has ~120 services.  say half of those are monitored on each device and you have several hundred devices.  that's a lot of unnecessary computations.